google/cloud/forseti/common/gcp_api/admin_directory.py

```diff
@@ -258,3 +261,120 @@ def get_users(self, customer_id='my_customer'):
             raise e
         except (errors.HttpError, HttpLib2Error) as e:
             raise api_errors.ApiExecutionError('users', e)
+
+
+class GroupSettingsClient(object):
+    """GSuite Admin Directory API Client."""
+
+    def __init__(self, global_configs, **kwargs):
+        """Initialize.
+
+        Args:
+            global_configs (dict): Global configurations.
+            **kwargs (dict): The kwargs.
+        """
+        credentials = api_helpers.get_delegated_credential(
+            global_configs.get('domain_super_admin_email'),
+            REQUIRED_SCOPES)
+
+        max_calls, quota_period = api_helpers.get_ratelimiter_config(
+            global_configs, API_NAME)
+
+        self.repository = GroupSettingsRepositoryClient(
+            credentials=credentials,
+            quota_max_calls=max_calls,
+            quota_period=quota_period,
+            use_rate_limiter=kwargs.get('use_rate_limiter', True))
+
+    def get_group_settings(self, customer_id='my_customer'):
+        """Get all the group settings for a given customer_id.
+
+        A note on customer_id='my_customer'. This is a magic string instead
+        of using the real customer id. See:
+
+        https://developers.google.com/admin-sdk/directory/v1/guides/manage-groups#get_all_domain_groups
+
+        Args:
+            customer_id (str): The customer id to scope the request to.
+
+        Returns:
+            list: A list of group settings objects returned from the API.
+
+        Raises:
+            api_errors.ApiExecutionError: If groups retrieval fails.
+            RefreshError: If the authentication fails.
+        """
+        try:
+            # self.repository.group_settings._get_key_field="groupUniqueId"
+            paged_results = self.repository.group_settings.get(customer_id)
```

Comment:
@blueandgold this is where error comes up complaining of invalid id

Reply:
This API doesn't return all group settings for the customer, it returns the settings for a specific group, it takes a group Unique ID (group email address) and returns the settings:

https://developers.google.com/apis-explorer/#search/groupssettings.groups.get/m/groupssettings/v1/groupsSettings.groups.get

Reply:
Andrew is correct, this is exactly the problem.  Please see PR #2539 to see how this works, using the group id in email format.

You can also try this out in the API Explorer.

Thanks Andrew for the help here!

Reply:
You can also see this in the API explorer.
![image](https://user-images.githubusercontent.com/6977544/52518698-8cc8ba80-2c03-11e9-8d34-b0e92bcd6d27.png)